### ISSUE TITLE
Description Content Can Contain Delimiters

### ISSUE THREAD

**Issue by nietonfir** [matiasperrone/ical-easy-reader#10]
Consider the following VEVENT with a dummy description of `a&b;c&d`:
```
BEGIN:VEVENT
CREATED:20210715T093624Z
LAST-MODIFIED:20210715T093633Z
DTSTAMP:20210715T093633Z
UID:afd8be94-4161-435e-8d91-6c117977790b
SUMMARY:FooBar
DTSTART;TZID=Europe/Anon:20210715T120000
DTEND;TZID=Europe/Anon:20210715T130000
TRANSP:OPAQUE
DESCRIPTION:a&b;c&d
END:VEVENT
```

When loading this VEVENT, the description is being split apart in `processMultivalue` because of the semicolon used. I think this is a bug, as I would expect the description not to be manipulated at all. So instead of `['a&b' => 'a&b', 'c&d' => 'c&d']` being returned in `['VEVENT'][0]['SUMMARY']` the original unaltered string should be returned: `a&b;c&d`.

**Comment by nietonfir** [comment]
The most obvious solution would be to simply skip processing the description string at all, e.g.:
```diff
--- iCalEasyReader.php.old	2021-07-15 13:29:54.486954158 +0200
+++ iCalEasyReader.php	2021-07-15 13:27:31.252638484 +0200
@@ -153,7 +153,7 @@
 		}
 
 		// Multi value
-		if (is_string($value)) {
+		if (is_string($value) && $key != 'DESCRIPTION') {
 			$this->processMultivalue($value);
 		}
 
```

But maybe you can come up with a better solution, as I'm not very experienced with the CalDAV format.

**Comment by matiasperrone** [comment]
Alright 3.1.6 will have the fix

**Comment by matiasperrone** [comment]
I think that SUMMARY has the same effect

**Comment by matiasperrone** [comment]
I am going to change the way is interpreted for only the keys that make sense to do this proceedings (call processMultivalue)

**Comment by nietonfir** [comment]
Sounds good! If you need help or anything, just ping. ;-)
And thank you for the quick response!

**Comment by matiasperrone** [comment]
Reading the documentation:
https://datatracker.ietf.org/doc/html/rfc5545#section-3.8.1.5 (DESCRIPTION)
https://datatracker.ietf.org/doc/html/rfc5545#section-3.3.11 (text data type)

ESCAPED-CHAR = ("\\" / "\;" / "\," / "\N" / "\n")
          ; \\ encodes \, \N or \n encodes newline
          ; \; encodes ;, \, encodes ,
          
So, it must be escaped.

**Comment by matiasperrone** [comment]
Issue fixed on v3.1.7 - Escaped caracters are correctly handled.
